metadata_backend/api/middlewares.py

```diff
@@ -25,22 +27,31 @@ async def http_error_handler(req: Request, handler: Callable) -> Response:
             response = await handler(req)
             return response
         except HTTPError as error:
-            return _json_exception(error.status, error)
+            return _json_exception(error.status, error, req.url)
 
     return http_error_handler
 
 
-def _json_exception(status: int, exception: Exception) -> Response:
-    """Convert an HTTP exception into JSON response.
+def _json_exception(status: int, exception: HTTPException,
+                    url: URL) -> Response:
+    """Convert an HTTP exception into a problem detailed JSON response.
+
+    The problem details are in accordance with RFC 7807.
+    (https://tools.ietf.org/html/rfc7807)
 
     :param status: Status code of the HTTP exception
     :param exception: Exception content
-    :returns: Response in JSON format
+    :param url: Request URL that caused the exception
+    :returns: Response in problem detail JSON format
     """
     body = json.dumps({
-        'error': status,
-        'detail': str(exception)
+        'type': "about:blank",
```

Comment:
maybe add link to swagger api schema here? It describes api responses a bit better than plain about:plank

Reply:
The type key is also optional so it could be left out for now. And if it's not included, its value is assumed to be 'about:blank'. The swagger api schema presents all possible errors so I'm not sure if it's that useful to include when describing a specific error.

Reply:
I think `about:blank` is ok for now until we will get a proper error schema.